src/annotator/pdf-sidebar.js

```diff
@@ -12,8 +16,101 @@ const defaultConfig = {
   },
 };
 
+// In accordance with the minimum width PDFJS asserts for its #mainContainer
+// element, we shouldn't try to shrink down the PDF container any narrower
+// than this value when resizing it in side-by-side mode. If the available space
+// left in the viewport after alloting space for the sidebar is less than this
+// value, don't apply side-by-side mode, but instead allow the sidebar to
+// overlap the PDF
+const MIN_PDF_WIDTH = 320;
+
 export default class PdfSidebar extends Sidebar {
   constructor(element, config) {
     super(element, { ...defaultConfig, ...config });
+
+    this.lastSidebarLayoutState = {
+      expanded: false,
+      width: 0,
+      height: 0,
+    };
+
+    this.window = /** @type {HypothesisWindow} */ (window);
+    this.pdfViewer = this.window.PDFViewerApplication?.pdfViewer;
+    this.pdfContainer = this.window.PDFViewerApplication?.appConfig?.appContainer;
+
+    // Prefer to lay out the sidebar and the PDF side-by-side (with the sidebar
+    // not overlapping the PDF) when space allows
+    this.sideBySide =
+      config.enableExperimentalPDFSideBySide &&
+      this.pdfViewer &&
+      this.pdfContainer;
+
+    // Is the current state of the layout side-by-side?
+    this.sideBySideActive = false;
+
+    if (this.sideBySide) {
+      this.subscribe('sidebarLayoutChanged', state =>
+        this.fitSideBySide(state)
+      );
+      window.addEventListener('resize', () => this.fitSideBySide());
```

Comment:
Do we need to debounce this?

Reply:
Probably, but not essential for a first iteration.